### ISSUE TITLE
dockerfile issue: sudo: command not found

### ISSUE THREAD

**Issue by xxlatgh** [martinbenson/deep-photo-styletransfer#6]
Hi Martin,
Thank you for providing the dockerfile! 

I was getting the following error messages when i tried to build the docker image.
install-deps: line 155: sudo: command not found                                                  
Some portion of the update is failed 
install-deps: line 161: sudo: command not found                                                  
The command '/bin/sh -c git clone https://github.com/torch/distro.git ~/torch --recursive &&     
cd ~/torch &&     bash install-deps &&     ./install.sh' returned a non-zero code: 127  

Then I fixed it by adding the following code after # Install dependencies
RUN apt-get update && \       
            apt-get -y install sudo

Not perfect but it works, perhaps you can take a look at this issue?

**Comment by martinbenson** [comment]
This seems strange, since the image builds fine for me as is and also everything runs as root anyway, so sudo is not needed?

**Comment by Multiboxer** [comment]
This happened to me when I ran the dockerfile through zsh instead of bash, quick fix was to switch to bash with `exec bash` after that it worked perfectly.

**Comment by martinbenson** [comment]
Thanks @Multiboxer  - do you mean if you try to run docker build from zsh it fails?

**Comment by xxlatgh** [comment]
@martinbenson I'm having encountering a few problems in the final step of building the dockerfile. 
First I had to change the makefile from cudo-7.5 to cuda-8.0 because i have previously installed cuda-8.0, now it's telling me that it can not find lua.h. Since it takes me more than 1 hr to build the image, I wonder if you can give me some pointers to speeed up my debugging. Thank you!

```
Step 17/17 : RUN make clean && make && pip3 install -r gen_laplacian/requirements.txt
 ---> Running in cd0552935bbc
find . -type f | xargs -n 5 touch
rm -f libcuda_utils.so  
/usr/local/cuda-8.0/bin/nvcc -arch sm_35 -O3 -DNDEBUG --compiler-options '-fPIC' -o libcuda_utils.so --shared cuda_utils.cu -I/home/ubuntu/torch/install/include/THC -I/home/ubuntu/torch/install/include/TH -I/home/ubuntu/torch/install/include -L/home/ubuntu/torch/install/lib -Xlinker -rpath,/home/ubuntu/torch/install/lib -lluaT -lTHC -lTH -lpng
cuda_utils.cu:2:18: fatal error: lua.h: No such file or directory
compilation terminated.
make: *** [libcuda_utils.so] Error 1
```

**Comment by alashow** [comment]
My build failed too, with `install-deps: line 161: sudo: command not found` error. Switching to `bash ` from `zsh` didn't help.

**Comment by xxlatgh** [comment]
hey @martinbenson, I fixed my earlier error, but i'm having problem with the final line in the dockerfile. Can you give me some pointers on how to fix this?

**What I've tried:**
breaking the last line in the docker file into 3 lines
`RUN make clean`
`RUN make `
`RUN pip3 install -r gen_laplacian/requirements.txt`

**What I found:**
bug is in Step 19/20 : RUN make
`Step 19/20 : RUN make
 ---> Running in 2f83ba0f5407
/usr/local/cuda-8.0/bin/nvcc -arch sm_35 -O3 -DNDEBUG --compiler-options '-fPIC' -o libcuda_utils.so --shared cuda_utils.cu -I/home/ubuntu/torch/install/include/THC -I/home/ubuntu/torch/install/include/TH -I/home/ubuntu/torch/install/include -L/home/ubuntu/torch/install/lib -Xlinker -rpath,/home/ubuntu/torch/install/lib -lluaT -lTHC -lTH -lpng
cuda_utils.cu:2:18: fatal error: lua.h: No such file or directory
compilation terminated.
makefile:10: recipe for target 'libcuda_utils.so' failed
make: *** [libcuda_utils.so] Error 1
The command '/bin/bash -c make' returned a non-zero code: 2`

**Comment by martinbenson** [comment]
Hi folks. Apologies, there were two problems with it:

I pulled the cuda image from dockerhub ages ago and they updated it to ubuntu 16 underneath me. So, I've just specified the ubuntu 14 image since it works.
I hacked the upstream makefile and copied in during the build and then forgot to add the edited makefile. I've fixed this properly by just editing the makefile in the fork

I'm hopeful this will resolve everyone's issues - but do not that you'll need recent nvidia drivers and nvidia-docker.
